### PR TITLE
User authentication

### LINKED ISSUE #7: Setup User authentication system
## Description:

In addition to the models that needs to be create and views and serializers, below are some endpoints to be considered.

Implement the following API endpoints to create a comprehensive user authentication system:

1. Registration:

        Endpoint: /api/register
        Method: POST
        Request Body:
        username: string (required)
        email: string (required, unique)
        password: string (required, securely hashed)
        Response:
        Status code: 201 (Created) if successful
        User object (optional): containing ID, username, email, etc.
        Error message (optional): if registration fails
2. Login:

        Endpoint: /api/login
        Method: POST
        Request Body:
        username or email: string (required)
        password: string (required)
        Response:
        Status code: 200 (OK) if successful
        Authentication token (e.g., JWT): for subsequent requests
        Error message (optional): if login fails
3. Password Reset:

        Endpoint: /api/password-reset
        Method: POST
        Request Body:
        email: string (required)
        Response: Password reset email sent (if user exists)
4. Password Reset Confirmation:

        Endpoint: /api/password-reset/{token}
        Method: PUT
        Request Body:
        password: string (required)
        password_confirmation: string (required)
        Response: Password updated (if token is valid)
5. User Profile:

        Endpoint: /api/profile
        Method: GET (protected)
        Response: User's profile information
6. Update Profile:

        Endpoint: /api/profile
        Method: PUT (protected)
        Request Body: Updated profile information
        Response: Updated profile
7. Logout:

        Endpoint: /api/logout
        Method: POST (protected)
        Response: Invalidates authentication token

### PR DESCRIPTION
**Pull request by LL-Etiane**
## Summary

Closes #7 
user authentication endpoints

### CODE REVIEW

File: apps/authentication/routes/__init__.py [hunk 1]
```diff
@@ -1 +0,0 @@
-��
```
Comment (yokwejuste) [inline apps/authentication/routes/__init__.py:1]: Are these files having the right encodings?

File: apps/users/serializers/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+from .auth_serializers import UserRegistrationSerializer, LoginSerializer, LoginResponseSerializer, PassWordResetRequestSerializer, PasswordResetConfirmationSerializer
```
Comment (yokwejuste) [inline apps/users/serializers/__init__.py:1]: We are normalizing full import statements.

```suggestion
from apps.authentication.auth_serializers import UserRegistrationSerializer, LoginSerializer, LoginResponseSerializer, PassWordResetRequestSerializer, PasswordResetConfirmationSerializer
```

So let's adopt the lines above and follow same for the rest.

**Review by yokwejuste — CHANGES_REQUESTED** [review]

File: apps/authentication/routes/__init__.py (hunk 1 above)
Comment (LL-Etiane) [reply apps/authentication/routes/__init__.py:1]: yes. Should be the action runner which was there

**Review by yokwejuste — APPROVED** [review]
LGTM 🌈

**Pull request merged**